### PR TITLE
Update Jenkinsfile

### LINKED ISSUE #4730: NUM_CORES should be 4 with new EKS runners
Right now the maximum number of cores allowed by AWS kubernetes is 4, so Jenkinsfile should reflect that.

https://github.com/USGS-Astrogeology/ISIS3/blob/0ec76e9b9e6ad09fd63ab8239fa9cc88c77fc7b1/Jenkinsfile#L3

### LINKED ISSUE #4731: Move channel_alias in Jenkinsfile
This has been moved to conda.prod-asc.chs.usgs.gov and builds will likely fail until the Jenkinsfile is updated.

https://github.com/USGS-Astrogeology/ISIS3/blob/0ec76e9b9e6ad09fd63ab8239fa9cc88c77fc7b1/Jenkinsfile#L37

### PR DESCRIPTION
**Pull request by jlaura**
Fixes #4730 to to set the max cores to 4 (the AWS EKS max) and fixes #4731 to update the conda cache URL

## Description

Modifications to the Jenkinsfile to reflect the move from local runners to remote runners on cloud hardware.

## Related Issue

#4730 and #4731

## Motivation and Context

Local hardware is being deprecated, so the these changes are needed to use the cloud infrastructure.

## How Has This Been Tested?

On merge this will execute and be tested.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/Code-Of-Conduct.md)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.
- [ ] I have added any user impacting changes to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

**Review by AustinSanders — DISMISSED** [review]

**Comment by jlaura** [comment]
@AustinSanders This is ready for a re-review. The CI is running now and building. So, the changes that I made are working as expected. No idea if the tests will pass or not, but this PR is updating the number of CPUs and the conda registry.

My previous attempt failed because I did not specify the http protocol on the registry URI.

**Review by acpaquette — COMMENTED** [review]
Only getting 4 cores has really slowed this down :/ A few of the module and app tests are timing out which happened with some regularity on 8 cores. We should double the timeout time and see if things pass

**Comment by jlaura** [comment]
I used the --timeout flag in the Jenkinsfile because (1) I don't see TIMEOUT being defined in our CMAKELISTS.txt files and (2) this encapsulates where the timeouts as being specific to our CI infrastructure.

**Comment by jlaura** [comment]
@acpaquette @jessemapel @ladoramkershner Did something change in our code base over the last few days with cnetcombinept? I'm not seeing any changes, but want to confirm. The build is now failing due to nanoflann, which this PR has nothing to do with.

**Comment by jessemapel** [comment]
The change history on cnetcombinept shows no changes since last February. It looks like we got hit with a new version of nanoflann. My current build environment has 1.3.2 but the CI is getting 1.4.0. We'll need to pin that back to <1.4 in the environment.yml.

File: environment.yml [hunk 1]
```diff
@@ -40,7 +40,7 @@ dependencies:
   - mesalib
   - mysql
   - mysql-connector-c
-  - nanoflann
+  - nanoflann<1.4
```
Comment (jessemapel) [inline environment.yml:43]: This also needs to be updated in the meta.yml recipe. I don't think we need to worry about the run pin, but the build pin needs to be updated.

**Comment by jlaura** [comment]
So the build failed with a 50 minute timeout. This seems, frankly ridiculous that these test need that much time. What are folks thoughts on disabling these tests and adding fixing them to the next support sprint? Alternatively, I'll double the timeout again to 6000s and re-run.

**Comment by jessemapel** [comment]
@jlaura What's the impact for bumping the timeout higher? I'd much prefer to live with longer CI runs for a little while than losing test coverage. Either way, moving these tests to gtest and/or smaller data sets seems like a high priority. For the record here's the tests timing out:

- 965 - juno_module_test_junocam
- 1086 - hijitter_app_test_default
- 1087 - hijitter_app_test_withCrop
- 1676 - ApolloCube.FunctionalTestApollocalDefault
- 1696 - TempTestingFiles.FunctionalTestApollopanstitcherDefault

**Comment by jessemapel** [comment]
Fixes #4742

**Review by AustinSanders — APPROVED** [review]

**Pull request merged**